src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt

```diff
@@ -84,6 +85,24 @@ class ADRController(private val encryption: Encryption,
     @GetMapping("/datasets")
     fun getDatasets(@RequestParam showInaccessible: Boolean = false): ResponseEntity<String>
     {
+        return SuccessResponse(listDatasets(showInaccessible).filter { it["resources"].count() > 0 }).asResponseEntity()
+    }
+
+    @GetMapping("/datasetsWithResource")
+    fun getDatasetsWithResource(@RequestParam resourceType: String, @RequestParam showInaccessible: Boolean = false): ResponseEntity<String>
+    {
+        val data = listDatasets(showInaccessible)
+        val fileTypeMappings = getApiToFileTypeMappings()
+        val filteredData = data.filter { dataset ->
+            dataset["resources"].any { resource ->
+                resource["resource_type"]?.asText() == fileTypeMappings[resourceType]
+            }
+        }
+
```

Comment:
@EmmaLRussell we chatted about if the ADR can do this for us. I think from looking - at the moment they can't. Similarly I thought we could pass a filter in to the initial `package_search` request to filter on only those with a resource but I can't get that to work. Even though the docs seem to suggest it should, so I am doing the fetch and filtering in kotlin after the fact.